api/src/main/java/org/eclipse/microprofile/config/Config.java

```diff
@@ -201,6 +201,51 @@
         return getOptionalValue(propertyName, arrayType).map(Arrays::asList);
     }
 
+    /**
+     * Return the resolved configuration properties instance with the specified prefix. 
+     * The type declaration can be annotated with 
+     * {@linkplain org.eclipse.microprofile.config.inject.ConfigProperties @ConfigProperties}. 
+     * If the type is annotated with {@linkplain org.eclipse.microprofile.config.inject.ConfigProperties @ConfigProperties}, 
+     * the prefix supplied in this method overrides the prefix associated with the annotation 
+     * {@linkplain org.eclipse.microprofile.config.inject.ConfigProperties @ConfigProperties}.
+     * 
+     * @param <T> 
+     *              The Class Type
+     * @param configProperties
+     *              The class that contains a number of fields that maps to corresponding configuration properties. 
+     *              The type declaration can be annotated with 
+     *              {@linkplain org.eclipse.microprofile.config.inject.ConfigProperties @ConfigProperties}.
+     *              The prefix as the method parameter overrides the prefix set on the type.                
+     *              This class should contain a zero-arg constructor. Otherwise, Otherwise, non-portable behaviour occurs.
+     * @param prefix
+     *              The prefix for the configuration properties declared on the class configProperties.
+     *              If the prefix is "", which means no prefix involved when performing property lookup.
+     *              If the prefix is null, this method is equivalent to {@linkplain #getConfigProperties(Class)}.
+     * @return      An instance for the specified type and prefix 
+     */
+    <T> T getConfigProperties(Class<T> configProperties, String prefix);
```

Comment:
Considering the last sentence of the javadoc:

> If the prefix is null, this method is equivalent to {@linkplain #getConfigProperties(Class)}.

Shouldn't ` <T> T getConfigProperties(Class<T> configProperties);` be implemented as a default method that calls `getConfigProperties(Class<T> configProperties, null)`?

Reply:
yes